crates/cdk/src/nuts/nut02.rs

```diff
@@ -114,14 +114,33 @@ impl Id {
     }
 }
 
-impl TryFrom<Id> for u64 {
+impl TryFrom<Id> for u32 {
     type Error = Error;
     fn try_from(value: Id) -> Result<Self, Self::Error> {
         let hex_bytes: [u8; 8] = value.to_bytes().try_into().map_err(|_| Error::Length)?;
 
         let int = u64::from_be_bytes(hex_bytes);
 
-        Ok(int % (2_u64.pow(31) - 1))
+        let result = (int % (2_u64.pow(31) - 1)) as u32;
+        Ok(result)
+    }
+}
+
+impl TryFrom<u64> for Id {
+    type Error = Error;
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let bytes = value.to_be_bytes();
+        Self::from_bytes(&bytes)
+    }
+}
+
+impl TryFrom<Id> for u64 {
+    type Error = Error;
+
+    fn try_from(value: Id) -> Result<Self, Self::Error> {
+        let bytes = value.to_bytes();
+        let byte_array: [u8; 8] = bytes.try_into().map_err(|_| Error::Length)?;
+        Ok(u64::from_be_bytes(byte_array))
```

Comment:
I'm hesitant to put in these u64 conversion and have the u32 one that does the module as defined in the nut. I think it maybe best to change the current `TryFrom<Id> for u64` to `TryFrom<Id> for u32` as you've done but to leave the u64 conversion out. Converting to and from u64 can be done outside cdk if thats needed for something as cdk should be kept to whats defined in the nuts.

Reply:
I see now I broke some other stuff with this change to u32. It's definitely good to isolate this change.

Reply:
don't merge, nut13 unit test incoming...